### ISSUE TITLE
Add ability to treat attachments as separate items

### ISSUE THREAD

**Issue by programmerq** [eikek/docspell#2063]
I was hoping to import documents that I receive via email. Sometimes multiple documents can come in the same email message, but they are separate documents.

One use case example is scraping the USPS informed delivery email messages I receive. They attach images of the mail that I will get for a given day.

This feels similar to https://github.com/eikek/docspell/pull/1459 where this sort of thing was made possible for zip files.

**Comment by eikek** [comment]
Hi @programmerq the same thing is possible for zip and eml files. But I think not from the "scan mailbox" task - do you mean this? If you upload an eml file with the option from #1459 you should get different items for mailbody and attachments.

**Comment by programmerq** [comment]
Ah yes, I should have specified that I was referring to the "scan mailbox" feature, but I have also seen this same behavior with the other methods for importing emails:

* `dsc watch --attachments-only --flatten-archives ...` didn't split them out. Same for `dsc upload`.
* I also used the`/app/upload` page, and checked the "Extract zip file contents into separate items, in contrast to a single document with multiple attachments." box. That also didn't split the attachments out.

The upload page does say "Choose document files (pdf, docx, txt, html, …). Archives (zip and eml) are extracted." in the user interface.

**Comment by eikek** [comment]
Ah, thanks for the feedback! Yes, then this doesn't work as I thought. I can't remember why it is not implemented for E-Mails, it should I'd think.

**Comment by madduck** [comment]
It's possible to add new files (attachments) to documents, maybe it would be helpful to also allow existing files to be promoted to their own documents?

**Comment by eikek** [comment]
> It's possible to add new files (attachments) to documents, maybe it would be helpful to also allow existing files to be promoted to their own documents?

Yes, that makes sense. I thought there is already an issue, but couldn't find one. I created #2261 now.

**Comment by eikek** [comment]
Just for context, I misunderstood the intention when originally reading this issue. Currently this is only possible for zip files, if the `flattenArchives` option is set to `true`. It works by extracting the zip and submitting jobs for each entry. This could be extended to work for eml files, but as it is now, the overall eml file would be removed. I think this is ok, since the body + attachments are stored as single files, but you wouldn't be able to retrieve the original file. That is why I didn't implemenet it for eml files back then.
